superset-frontend/src/visualizations/FilterBox/FilterBox.jsx

```diff
@@ -336,10 +328,12 @@ class FilterBox extends React.PureComponent {
     // Add created options to filtersChoices, even though it doesn't exist,
     // or these options will exist in query sql but invisible to end user.
     Object.keys(selectedValues)
-      .filter(
-        key => selectedValues.hasOwnProperty(key) && key in filtersChoices,
-      )
+      .filter(key => key in filtersChoices)
       .forEach(key => {
+        // empty values are ignored
+        if (!selectedValues[key]) {
+          return;
+        }
```

Comment:
This will fix this bug:

### Before

<img width="382" alt="has-empty-value" src="https://user-images.githubusercontent.com/335541/107862200-0d78ad80-6e00-11eb-9771-ae67c11058f5.png">

### After

<img width="386" alt="no-empty-value" src="https://user-images.githubusercontent.com/335541/107862203-12d5f800-6e00-11eb-8ef3-303fea3d56ee.png">